### PR TITLE
Ensure network connections are restored after disruptions

### PR DESCRIPTION
**Pull request by bleskes**
With #22977, network disruption also disconnects nodes from the transport service. That has the side effect that when the disruption is healed, the disconnected node stay disconnected until the `NodeConnectionsService` restores the connection. This can take too long for the tests. This PR adds logic to the cluster healing to restore connections immediately. 

See https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian/611/console for an example failure.

### CODE REVIEW

**Comment by bleskes** [comment]
Test this please

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 1]
```diff
@@ -1169,6 +1169,21 @@ protected void ensureStableCluster(int nodeCount, TimeValue timeValue, boolean l
...
+
+    /**
+     * Ensures that all nodes in the cluster are connected to each other.
+     *
+     * Some network disruptions may leave nodes that are not the master disconnected from each other.
+     * {@link org.elasticsearch.cluster.NodeConnectionsService} will eventually reconnect but it's
+     * handy to be able to ensure this happens faster
+     */
+    protected void ensureFullyConnectedCluster() {
+        for (String node: internalCluster().getNodeNames()) {
```
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: Can you add a method to the test framework that takes an `InternalCluster` and executes this exact logic? You can invoke it here, and in `NetworkDisruption`.

**Review by jasontedor — CHANGES_REQUESTED** [review]
Looks good, but I left one request.

**Comment by bleskes** [comment]
Thx @jasontedor . I pushed another commit.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**